core/client/hdfs3/pom.xml

```diff
@@ -0,0 +1,53 @@
+<!--
+    The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+    (the "License"). You may not use this work except in compliance with the License, which is
+    available at www.apache.org/licenses/LICENSE-2.0
+    This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+    either express or implied, as more fully set forth in the License.
+    See the NOTICE file distributed with this work for information regarding copyright ownership.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.alluxio</groupId>
+    <artifactId>alluxio-core-client</artifactId>
+    <version>2.10.0-SNAPSHOT</version>
+  </parent>
+  <artifactId>alluxio-core-client-hdfs3</artifactId>
```

Comment:
why not
add a `hadoop-3` profile in core/client/hdfs/pom.xml with extra needed dependencies

Reply:
> why not add a `hadoop-3` profile in core/client/hdfs/pom.xml with extra needed dependencies



> > > Do we really need symbolic link, and does this mean that we need to generate two release tarball in the end.
> > 
> > 
> > the symbolic link avoids the need to create two release tarballs. since there are 2 types of client jars, we could choose to publish two full release tarballs with 1 client jar each. the alternative is to publish one full release tarball with both client jars, but because only one client jar can be on the classpath, the symbolic link points to the designated client jar. if there is a need to switch to the other client jar, it only requires a modification of a symlink, rather than downloading a separate release tarball.
> 
> My understanding is that there are two more tarballs that need to be released, one linked to the hadoop2 client and one linked to the hadoop3 client. My implementation in #16017 does not require two tarballs, and applications are still free to choose whether to use hadoop2 or hadoop3 clients in their classpath.

i think there may still be some misunderstanding; my code is identical to yours except i add a profile option to overwrite the symlink that is the default location for the alluxio client jar. both client tarballs will be built and included within the tarball (this is the same as yours), but we have some control which client jar is used by default in the release tarball.

Reply:
@Xenorith Just trying to see if we can avoid the copy of that two pom files, and use profile to dynamically build.
Agree that we can have two separate jars for hadoop 2 and 3. But that can be executed when generating tarballs with extra maven build command to generate the additional hadoop version har

Reply:
oh if you know of a better way to use the profile. currently this PR shows that there is no code diff, but keep in mind that we intend of making code changes between hadoop2 and hadoop3, so i believe we will still need separate pom files for separate projects; see https://github.com/Alluxio/alluxio/pull/16017/files#diff-d26119e0235b0e93f1493971a0472543e5c7ee3d7e5d92c0c7fb22c860a83770

Reply:
oh got it, if code diff is needed, then separate pom files are unavoided